nucypher/cli/characters/stake.py

```diff
@@ -290,10 +297,44 @@ def stake(click_config,
                                             transactions=new_stake.transactions)
         return  # Exit
 
+    elif action == "restake":
+
+        # Authenticate
+        if not staking_address:
+            staking_address = select_stake(stakeholder=STAKEHOLDER, emitter=emitter).staker_address
+        password = None
+        if not hw_wallet and not blockchain.client.is_local:
+            password = get_client_password(checksum_address=staking_address)
+        STAKEHOLDER.assimilate(checksum_address=staking_address, password=password)
+
+        # Inner Exclusive Switch
+        if lock:
+            if not force:
+                click.confirm(f"Confirm enable restaking lock for staker {staking_address} until {release_period}?", abort=True)
```

Comment:
It seems that we are using `release_period` before prompting for it if needed (see lines below)...

Perhaps this should be moved after the `release_period` value is checked/determined

Reply:
:heavy_check_mark: